### PR TITLE
Optimise the performance of lesson plan edit page

### PR DESCRIPTION
**Pull request by allenwq**
The slowness is because all items in the page get re-rendered if there's any changes.

Moved `renderItem` to a separate component and implemented `shouldComponentUpdate` to avoid unnecessary re-render.

### CODE REVIEW

File: client/app/bundles/course/lesson_plan/components/LessonPlanItemEdit.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  static propTypes = {
+    item: PropTypes.instanceOf(Immutable.Map).isRequired,
+    updateItem: PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  shouldComponentUpdate(nextProps) {
+    const nextItem = nextProps.item;
+    const item = this.props.item;
+    if (nextItem.get('start_at') !== item.get('start_at') ||
```
Comment (weiqingtoh) [inline client/app/bundles/course/lesson_plan/components/LessonPlanItemEdit.jsx]: Just wondering if there a neater way to write this?

Comment (allenwq) [inline client/app/bundles/course/lesson_plan/components/LessonPlanItemEdit.jsx]: Add a new commit and use the other way for comparison

**Comment by kxmbrian** [comment]
LGTM. Still pending to clean up lesson plan and edit page properly.

**Comment by allenwq** [comment]
So I squash there 2 commits ?  @weiqingtoh any comments about the second commit ?

**Comment by weiqingtoh** [comment]
LGTM!

**Pull request merged**